### PR TITLE
ESP32C3 stable.

### PR DESCRIPTION
**Pull request by abhishek2desh**
ESP32-C3-Devkit stable.

### CODE REVIEW

**Comment by blazoncek** [comment]
Why? I have C3 and it is running stable with current set-up.
I would prefer not to use external platforms if not necessary.

Other than that, making unnecessary changes in platformio.ini is not welcome.

File: platformio.ini [hunk 1]
```diff
@@ -10,7 +10,7 @@
 # ------------------------------------------------------------------------------
 
 # Release / CI binaries
-default_envs = nodemcuv2, esp8266_2m, esp01_1m_full, esp32dev, esp32_eth, esp32s2_saola, esp32c3, esp32s3dev_8MB
+# default_envs = nodemcuv2, esp8266_2m, esp01_1m_full, esp32dev, esp32_eth, esp32s2_saola, esp32c3, esp32s3dev_8MB
```
Comment (blazoncek) [inline platformio.ini:13]: Don't change that.

File: platformio.ini [hunk 2]
```diff
@@ -381,15 +382,19 @@ build_flags = ${common.build_flags} ${esp32s2.build_flags} #-D WLED_RELEASE_NAME
 lib_deps = ${esp32s2.lib_deps}
 
 [env:esp32c3]
-platform = espressif32@5.1.1
+platform = https://github.com/tasmota/platform-espressif32/releases/download/v2.0.3rc1/platform-espressif32-2.0.3new.zip
```
Comment (blazoncek) [inline platformio.ini:385]: Why not stock? Where is the issue with stock platform?

File: platformio.ini [hunk 3]
```diff
@@ -381,15 +382,19 @@ build_flags = ${common.build_flags} ${esp32s2.build_flags} #-D WLED_RELEASE_NAME
...
 framework = arduino
 board = esp32-c3-devkitm-1
 board_build.partitions = tools/WLED_ESP32_4MB_1MB_FS.csv
 build_flags = ${common.build_flags} ${esp32c3.build_flags} #-D WLED_RELEASE_NAME=ESP32-C3
   -D WLED_WATCHDOG_TIMEOUT=0
+  -D LEDPIN=2
```
Comment (blazoncek) [inline platformio.ini:391]: Not needed.

**Review by blazoncek — CHANGES_REQUESTED** [review]
If you want custom build environment use `platformio_override.ini` and do not change upstream platformio.ini.

**Comment by softhack007** [comment]
Sorry but I will not approve this PR as it is today. 
Reasons as explained by @blazoncek.

**Comment by abhishek2desh** [comment]
I tried using current setup but there were some issues with littlefs
library and the board kept rebooting indefinitely. So I made the change to
ini file.

**Comment by Aircoookie** [comment]
I would support adding the tasmota-compiled platform as a commented out option. The other changes to platformio will not get merged.

**Comment by softhack007** [comment]
Thanks @Aircoookie this solution seems to be a good compromise.

**Comment by softhack007** [comment]
I have been playing around with a -C3 recently, and also saw issues with corrupted littlefs, that happen with espressif@5.1.1 in combination with ARDUINO_USB_CDC_ON_BOOT=0. These problems disappear when using espressif@5.2.0. Espressif@5.1.1 seems to work when ARDUINO_USB_CDC_ON_BOOT=1. So my proposal is to try with espressif@5.2.0.

@abhishek2desh can you explain a bit more why you picked `platform=https://github.com/tasmota/platform-espressif32/releases/download/v2.0.3rc1/platform-espressif32-2.0.3new.zip` ? It seems to be a pre-release from tasmota, based on arduino-esp32 v2.0.3 in "tasmota flavour". What makes this one special?

**Comment by blazoncek** [comment]
@softhack007 indeed ARDUINO_USB_CDC_ON_BOOT caused many issues as well with S2 when I was testing it (corrupting LittleFS and/or rebooting/panicking).

**Comment by softhack007** [comment]
A few users have confirmed already that littlefs corruption on -C3 is resolved by using `platform=espressif@5.2.0`. So I prefer to have this as an optional line in platformio.ini, instead of an older tasmota pre-release version.

Closing this PR, in favour of commit https://github.com/Aircoookie/WLED/commit/b8cc78358367befc134f133738e2e58157ee92dd

**Comment by Paras-PC** [comment]
> Why? I have C3 and it is running stable with current set-up. I would prefer not to use external platforms if not necessary.
> 
> Other than that, making unnecessary changes in platformio.ini is not welcome.

Which file did you flash on your ESP-C3?
I want to flash a Seeed Studio XIAO ESP32C3

**Comment by blazoncek** [comment]
> Which file did you flash on your ESP-C3? I want to flash a Seeed Studio XIAO ESP32C3

My own compile from PIO.
Env posted on Discord.

**Pull request closed without merging**